python/lsst/afw/geom/span.cc

```diff
@@ -0,0 +1,95 @@
+/* 
+ * LSST Data Management System
+ * Copyright 2008-2016  AURA/LSST.
+ * 
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the LSST License Statement and 
+ * the GNU General Public License along with this program.  If not, 
+ * see <https://www.lsstcorp.org/LegalNotices/>.
+ */
+
+#include <pybind11/pybind11.h>
+#include <pybind11/operators.h>
+
+#include "lsst/afw/geom/Span.h"
+#include "lsst/afw/geom/SpanPixelIterator.h"
+
+namespace py = pybind11;
+
+using namespace lsst::afw::geom;
+
+// A thin wrapper around SpanPixelIterator.
+// Unfortunately we cannot use py::make_iterator here, as we normally
+// should, because for some reason the return values then all refer
```

Comment:
A different discussion with @pschella led me to look up the iterator category for `SpanPixelIterator`, and it is indeed incorrectly set to `random_access_iterator` rather than either `input_iterator` or `forward_iterator` (I haven't thought hard enough about this particular case to figure out which it ought to be yet).  I'm not certain fixing that will fix the problem; pybind11 would have to have different implementations for different iterator categories (presumably as an optimization) for this to matter, but it could be what's going wrong.  In any case, our code should be fixed, and it'd be worth seeing if that allows `py::make_iterator` to work properly.